### DISCUSSION TITLE
mini.map - minimap.gen_integration.builtin_search does not clear on "<c-l>"

### DISCUSSION THREAD

**Discussion by yavorski** [nvim-mini/mini.nvim#356] [Q&A] [answered]
Hi, when search is activated there are active highlights in the `mini.map`, however when you press `<c-l>` to clear the highlights, search highlights in `mini.map` are not cleared from the `mini.map` view/window, how can we achieve that? 

Here is my config using lazy

```lua
{
  "echasnovski/mini.map",
  keys = {{ "<leader>M", function() require("mini.map").toggle() end, silent = true, desc = "MiniMapToggle" }},
  config = function()
    local minimap = require("mini.map")

    minimap.setup({
      symbols = {
        encode = minimap.gen_encode_symbols.dot("4x2")
      },
      window = {
        width = 12,
        show_integration_count = false
      },
      integrations = {
        minimap.gen_integration.gitsigns(),
        minimap.gen_integration.diagnostic({
          info = "DiagnosticFloatingInfo",
          hint = "DiagnosticFloatingHint",
          warn = "DiagnosticFloatingWarn",
          error = "DiagnosticFloatingError"
        }),
        minimap.gen_integration.builtin_search(),
      },
    })
  end
}
```

Mini plugins are awesome, thank you for that!

**Comment by echasnovski** [comment, marked as answer]
> Hi, when search is activated there are active highlights in the `mini.map`, however when you press `<c-l>` to clear the highlights, search highlights in `mini.map` are not cleared from the `mini.map` view/window, how can we achieve that?

This is mostly because built-in mapping for `<C-l>` doesn't really update 'hlsearch' option. `builtin_search` tracking 'hlsearch' changes is mostly the only way I could find for its automated updates.

There are at least these approaches to solve the issue:
- Go the [route of 'mini.basics'](https://github.com/echasnovski/mini.nvim/blob/b1a6d9692e2f67ee35f77792b61eb7d44651c535/lua/mini/basics.lua#L615) and define right hand side of mapping as `<Cmd>let v:hlsearch = 1 - v:hlsearch<CR>`. It can be `<C-l>` or any other (like `\h` in 'mini.basics').
- Go the built-in route and use its command **but** starting with `:` instead of `<Cmd>`. This will change mode and prompt refresh of map:
```lua
vim.keymap.set('n', '<C-l>', ':nohlsearch | diffupdate | normal! <C-l><CR>')
```

> Mini plugins are awesome, thank you for that!

Thanks!

**Reply by yavorski** [reply]
Thanks!
